lib/Sema/ConstraintSystem.cpp

```diff
@@ -790,8 +791,7 @@ Type ConstraintSystem::openUnboundGenericType(
 
   type = type.transform([&](Type type) -> Type {
       if (auto unbound = type->getAs<UnboundGenericType>()) {
-        OpenedTypeMap replacements;
-        return openUnboundGenericType(unbound, locator, replacements);
+        return openUnboundGenericType(unbound, locator);
```

Comment:
Are you sure this is NFC?

Reply:
Based on the use it seems to be, replacements are local to a particular type...

Reply:
> Are you sure this is NFC?

It has to be — there is no recursion.